pandas/core/series.py

```diff
@@ -623,6 +623,29 @@ def view(self, dtype=None):
         return self._constructor(self._values.view(dtype),
                                  index=self.index).__finalize__(self)
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
```

Comment:
(How) should we handle the `out` parameter? Is that possible? On my WIP I had

```diff
+        if out:
+            kwargs['out'] = tuple(x._values if isinstance(x, cls) else x
+                                  for x in out)
```

but I don't know if that was tested.

Reply:
xarray is for example simply erroring in such a case:

https://github.com/pydata/xarray/blob/5ebed791db006075701c16b2e8eb9566207f2548/xarray/core/arithmetic.py#L53-L60

In the above, you would pass through an ExtensionArray as `out`, but that means that the ExtensionArray implementation of `out` should support it. 
But, maybe we should indeed do pass it through here (as you show), and let the `ExtensionArray.__array_ufunc__` error on an EA as `out` if it wants?

Reply:
Would it necessarily be an ExtensionArray, or could it also be a Series for `out`?

But raising seems fine for now.

Reply:
In master, an `out` argument that is not an array does not seem to be supported:

```
In [47]: s = pd.Series([1, 2, 3])

In [49]: out = s.copy()

In [50]: np.add(s, s, out=out)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-50-482c18e97629> in <module>()
----> 1 np.add(s, s, out=out)

TypeError: return arrays must be of ArrayType
```